src/classes/UTIL_Query_TEST.cls

```diff
@@ -116,6 +135,100 @@ private class UTIL_Query_TEST {
         System.assert(expectedAccountIds.containsAll(actualAccountIds), 'Expected Accounts ' + expectedAccounts + ' should be the same as actual Accounts ' + actualAccounts);
     }
 
+    /*********************************************************************************************************
+    @description 
+        Tests SOQL builder when multiple WHERE Clause are provided
+    verify:
+        The SOQL contains WHERE Clause
+    **********************************************************************************************************/
+    private static testMethod void testSoqlIsCorrectWhenMultipleWhereClausesAreProvided() {
+        Account[] accounts = new Account[]{ 
+            new Account(Name = 'Test1'),
+            new Account(Name = 'Test2'),
+            new Account(Name = 'Test3')
+        };
+        insert accounts;
+
+        String[] whereClauses = new String[]{
+            'Id IN :expectedAccounts',
```

Comment:
In this particular example, a string where clause is being passed into the Query builder. That where clause uses a bound variable (ex: `'Id IN :expectedAccounts'`). The query itself would be executed in the UTIL_Query (or maybe UTIL_Finder), however a query that uses a bound variable that is not in scope will fail. This test method doesn't explicitly check that condition, but my larger concern is that the `withWhere(String)` method is meant to accept a where clause fragment that can include a bound variable that will be out of scope when the query is executed. I hope this makes sense. Not 100% sure what the appropriate solution is. Could be to throw an exception with there is a `:` in the string. Open to discussion.

Reply:
@force2b - The build function returns the query string and the query is ran by the caller so scope should not be an issue here.

Reply:
The `UTIL_Finder` class has a `database.query()` in it and that method uses the `build()` method in `UTIL_Query` to generate the query. If you're confident that none of the existing code will pass a where clause string with a bound var I think it's fine, though an extra check for a bound var in the `UTIL_Finder.queryRecords()` method might help avoid an unexpected exception.

Reply:
The UTIL_Finder builds the where clause based on fields and values passed in to the UTIL_Where, so I feel confident we will not pass an a bound variable unintentionally.

Reply:
Thank you both for clarifying. I'm fine with leaving this as is given the usage.